### PR TITLE
[Dependency Update] CUDA10.1 Support

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
Upgrade the CUDA 10.1 with latest cuDNN **7.5.1** & NCCL **2.4.2**

## Checklist ##
Run three models ResNet50 with ImageNet & LSTM with PTB & MLP with MNIST
Performance shown below
Environment: P3.16xlarge Deep Learning Base AMI
Codebase: commit d87bd2aba1043d1f3c9e998d6ecd4bee2e342bae
The unit of thoughput is **samples/per second**
Each throughput is calcuated by average of 5 runs
### ResNet ###
**model**: Resnet50
**dataset**: Imagenet
**number of gpu**: 8
**epochs**: 90 (since the regression we found recently only have significant impact on large epochs)
**preprocess command**: sudo pip install gluoncv==0.2.0b20180625
**command**: python mxnet_benchmark/train_imagenet.py --use-rec --batch-size 128 --dtype float32 --num-data-workers 40 --num-epochs 3 --gpus 0,1,2,3,4,5,6,7 --lr 0.05 --last-gamma --mode symbolic --model resnet50_v1b --rec-train /home/ubuntu/data/train-passthrough.rec --rec-train-idx /home/ubuntu/data/train-passthrough.idx --rec-val /home/ubuntu/data/val-passthrough.rec --rec-val-idx /home/ubuntu/data/val-passthrough.idx
**github repo**: https://github.com/rahul003/deep-learning-benchmark-mirror.git

|  Throughput |  CUDA10.1 cuDNN 7.5.1/NCCL 2.4.2     | CUDA 10 cuDNN 7.5.1/NCCL 2.4.2 | Perforamnce Difference|
|:----------|:------------------------:|:--------------------:|:---------------------:|
|with MKLDNN | 2817.18815 | 2791.17069 | 0.932%  |
|without MKLDNN | 2821.88889 | 2798.57505 | 0.833%  |

### LSTM ###
**model**: LSTM
**dataset**: PTB(Penn Treebank)
**number of gpu**: 1
**epochs**: 10
**command**:
python2 benchmark_driver.py --framework mxnet --task-name mkl_lstm_ptb_symbolic --num-gpus 1 --epochs 10 --metrics-suffix test --kvstore local
python word_language_model/lstm_bucketing.py —num-hidden 650 —num-embed 650 —gpus 0 --epochs 10 --kv-store local

| Throughput   |  CUDA10.1 cuDNN 7.5.1/NCCL 2.4.2     | CUDA 10 cuDNN 7.5.1/NCCL 2.4.2 | Perforamnce Difference|
|:----------|:------------------------:|:--------------------:|:---------------------:|
| with MKLDNN | 1015.61785 | 869.05555(Performance Regression) | 16.865%  |
| without MKLDNN | 1015.01455 | 830.68338(Performance Regression) | 22.190%  |

**The CUDA 10 have a performance regression issue, please see #14725 to find more details.**

### MLP ###
I changed the MLP script so the performance might be a liitle worse than before
**model**: 3 dense layers with num_hidden=64 and relu as activation
**dataset**: MNIST
**number of gpu**: 1
**epochs**: 10
**command**:
python2 benchmark_runner.py —framework mxnet —metrics-policy mlp —task-name mlp —metrics-suffix test —num-gpus 1 —command-to-execute 'python3 mlp.py' —data-set mnist

| Throughput   |  CUDA10.1 cuDNN 7.5.1/NCCL 2.4.2     | CUDA 10 cuDNN 7.5.1/NCCL 2.4.2 | Perforamnce Difference|
|:----------|:------------------------:|:--------------------:|:---------------------:|
|  with MKLDNN | 3763.73161 | 3831.80019 | -1.776%  |
|  without MKLDNN | 4094.43442 | 3970.82162 | 3.226%  |

## Comments ##
@szha @lanking520 @eric-haibin-lin

### CODE REVIEW

File: tools/setup_gpu_build_tools.sh [hunk 1]
```diff
@@ -85,7 +91,31 @@ if [[ $VARIANT == cu* ]]; then
 fi
 
 # list of debs to download from nvidia
-if [[ $VARIANT == cu100* ]]; then
+if [[ $VARIANT == cu101* ]]; then
+    cuda_files=( \
```
Comment (szha) [inline tools/setup_gpu_build_tools.sh]: I added this before and here's what I got:
```
      "cuda-core-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "libcublas10_${CUDA_PATCH_VERSION}_amd64.deb" \
      "libcublas-dev_${CUDA_PATCH_VERSION}_amd64.deb" \
      "cuda-cudart-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-cudart-dev-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-curand-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-curand-dev-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-cufft-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-cufft-dev-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-nvrtc-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-nvrtc-dev-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-cusolver-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-cusolver-dev-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-misc-headers-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "cuda-nvcc-${CUDA_MAJOR_DASH}_${CUDA_VERSION}_amd64.deb" \
      "libcuda1-${LIBCUDA_MAJOR}_${LIBCUDA_VERSION}_amd64.deb" \
      "nvidia-${LIBCUDA_MAJOR}_${LIBCUDA_VERSION}_amd64.deb" \
```

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: tools/setup_gpu_build_tools.sh (hunk 1 above)
Comment (stu1130) [reply tools/setup_gpu_build_tools.sh]: Thanks @szha I will rerun the benchmark with latest build script

**Comment by lanking520** [comment]
General thoughts, do you think it is nessary for us to have some real-time benchmarking on the performance once we do some upgrade like this?

**Comment by stu1130** [comment]
@lanking520 Sorry for late response. real-time benchmarking here means test performance in CI System? I would prefer running the bechmark across CUDA 9/9.2/10/10.1 on nightly build

**Comment by lanking520** [comment]
@perdasilva I saw a recent commit done by you to downgrade the CUDA version. Do you think this number is promising and we push it to master?

**Pull request merged**